lib/active_fedora/associations/collection_association.rb

```diff
@@ -367,7 +367,11 @@ def delete_or_destroy(records, method)
 
           records.each { |record| callback(:before_remove, record) }
           delete_records(existing_records, method) if existing_records.any?
-          records.each do |record|
+          # Existing_records are deleted in the line above, therefore
+          # I am excluding them from the loop below. This seems to be
+          # causing problems with 4 integration tests. Still need to
+          # figure out why.
+          (records-existing_records).each do |record|
```

Comment:
@jcoyne I updated this loop to only process records that have not already been deleted. Yet, this is now making 4 integration tests fail. 

Does this change make sense? I updated this loop because one of the new tests was throwing an LDP::Gone when it was looping to see if one of the records to delete was new. The LDP::Gone was because the record was deleted in the call to `delete_records` just above the loop. If this change make sense I can see why the integration tests are failing but wanted to check before I go down the rabbit hole.